### ISSUE TITLE
Missing dependency @aws-sdk/signature-v4-crt after updating to client-s3 3.33.0

### ISSUE THREAD

**Issue by birtles** [aws/aws-sdk-js-v3#2806]
## Describe the bug

After updating to `@aws-sdk/client-s3` 3.33.0 from 3.32.0 I get the following error when building lambdas using `NodejsFunction` as part of a CDK project:

```
$ cdk deploy --all
Bundling asset Stack/function/Code/Stage...
 > node_modules/@aws-sdk/middleware-sdk-s3/dist/cjs/S3SignatureV4.js: error: Could not resolve "@aws-sdk/signature-v4-crt" (mark it as external to exclude it from the bundle)
    38 │ ...erV4 = (await Promise.resolve().then(() => __importStar(require("@aws-sdk/signature-v4-crt")))).CrtSignerV4;
       ╵                                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~

1 error
```

#### SDK version number
`@aws-sdk/client-s3@3.33.0`

#### Is the issue in the browser/Node.js/ReactNative?
Node.js

#### Details of the browser/Node.js/ReactNative version
v14.17.5

## Additional context
Presumably a regression from #2796.

Reverting to 3.32.0 fixes the issue.

cc @AllanZhengYP

**Comment by dianababurina-photobox** [comment]
Facing the same issue with serverless + webpack

**Comment by 0xdeafcafe** [comment]
Same here.

**Comment by AllanZhengYP** [comment]
Hi @birtles @dianababurina-photobox @0xdeafcafe, this is indeed a regression introduced by https://github.com/aws/aws-sdk-js-v3/pull/2796. As it mentioned in the description, we currently don't have functional test agains esbuild used by CDK Node.js function component. We will need to enforce the test.

**Comment by bdo** [comment]
An alternative workaround (if you actually don't need S3 Multi-region Access Point), is to entirely **exclude** `@aws-sdk/signature-v4-crt` from the bundle.

For instance with serverless-bundle:
```yaml
custom:
  bundle:
    forceExclude:
      - '@aws-sdk/signature-v4-crt'
```

**Comment by henhal** [comment]
I commented on this issue believing that it had been unfixed, but this seems to be a new issue. 
 https://github.com/aws/aws-sdk-js-v3/issues/2750#issuecomment-925780191

It's becoming a bad habit that about every second release is broken. :/

**Comment by AllanZhengYP** [comment]
@henhal The incompatible issue with bundler has happened a couple of times. We are working on adding the test with popular bundlers: WebPack4, WebPack5, Rollup, EsBuild to prevent this from happening again.

**Comment by oliversalzburg** [comment]
Similarly in yarn, for additional keyword stuffing:
```shell
➤ YN0002: │ @aws-sdk/client-s3@npm:3.33.0 doesn't provide @aws-sdk/signature-v4-crt (p3df12), requested by @aws-sdk/middleware-sdk-s3
➤ YN0002: │ @aws-sdk/s3-request-presigner@npm:3.33.0 doesn't provide @aws-sdk/signature-v4-crt (p0c252), requested by @aws-sdk/middleware-sdk-s3
```

**Comment by oliversalzburg** [comment]
@AllanZhengYP I figured this should be fixed in 3.34 now, but even installing 3.35 still raises the same peer dependency warnings.
